From f61ec1dcf0d09f9d1c5b728eaea060fdd137db50 Mon Sep 17 00:00:00 2001
From: Nathan Barrett Morrison <nathan.morrison@timesys.com>
Date: Wed, 8 May 2019 12:41:16 -0400
Subject: [PATCH 3/3] Applied
 0003_cover_cache_check_for_MultiCore_MCAPI_demo_and_test.patch

---
 tests/arm_sharc_msg_demo.c | 4 +++-
 tests/arm_sharc_msg_test.c | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/tests/arm_sharc_msg_demo.c b/tests/arm_sharc_msg_demo.c
index f1dffd8..88d31fd 100644
--- a/tests/arm_sharc_msg_demo.c
+++ b/tests/arm_sharc_msg_demo.c
@@ -173,6 +173,7 @@ int main (int argc, char *argv[])
 	mcapi_info_t version;
 	mcapi_endpoint_t ep1,ep2;
 	char recv_buf[BUFF_SIZE];
+	char cmp_buf[BUFF_SIZE];
 	char send_buf[32] = "";
 	int s = 0, pass_num = 0;
 	size_t size;
@@ -277,7 +278,8 @@ int main (int argc, char *argv[])
 		send(ep1, ep2, send_buf, strlen(send_buf), &status, &request, mode, timeout);
 		printf("Core0: mode(%d) message send. The %d time sending\n", mode, s);
 		recv(ep1, recv_buf, sizeof(recv_buf)-1, &status, &request, mode, timeout);
-		if (!strncmp(recv_buf, RECV_STRING, sizeof(recv_buf))) {
+		snprintf(cmp_buf, sizeof(cmp_buf), "%s %d", RECV_STRING, s+1);
+		if (!strncmp(recv_buf, cmp_buf, sizeof(recv_buf))) {
 			pass_num++;
 			printf("Core0: mode(%d) message recv. The %d time receiving\n",mode, s);
 		} else
diff --git a/tests/arm_sharc_msg_test.c b/tests/arm_sharc_msg_test.c
index 2e5e558..6e8538e 100644
--- a/tests/arm_sharc_msg_test.c
+++ b/tests/arm_sharc_msg_test.c
@@ -213,6 +213,7 @@ int main (int argc, char *argv[])
 	char str[] = "123";
 	char send_buf[32] = "";
 	char recv_buf[BUFF_SIZE];
+	char cmp_buf[BUFF_SIZE];
 	int s = 0, pass_num = 0;
 	size_t size;
 	int mode = 0;
@@ -334,7 +335,8 @@ int main (int argc, char *argv[])
 		send(ep1, ep2, send_buf, strlen(send_buf), &status, &request, mode, timeout, delay_us);
 		printf("Core0: mode(%d) message send. The %d time sending\n",mode, s);
 		recv(ep1, recv_buf, sizeof(recv_buf)-1, &status, &request, mode, timeout, delay_us);
-		if (!strncmp(recv_buf, RECV_STRING, sizeof(recv_buf))) {
+		snprintf(cmp_buf, sizeof(cmp_buf), "%s %d", RECV_STRING, s+1);
+		if (!strncmp(recv_buf, cmp_buf, sizeof(recv_buf))) {
 			pass_num++;
 			printf("Core0: mode(%d) message recv. The %d time receiving\n",mode, s);
 		} else
-- 
2.11.0

